database.go

```diff
@@ -1,37 +1,81 @@
 package main
 
 import (
-	"database/sql"
+	"errors"
+	"fmt"
+	"strings"
 )
 
-var db *sql.DB
+var (
+	ErrDBConnStrParseNoSep = errors.New("Missing ':' in connection string")
+	ErrDBConnStrParseNotImplemented = func(dbName string) error {
+		return errors.New(fmt.Sprintf("The database '%s' is not implemented", dbName))
+	}
+)
 
-func LoadDatabase(dbFile string) error {
-	var err error
-	db, err = sql.Open("sqlite3", dbFile)
+type Database interface {
+	CommentService
+}
+
+func parseConnectionStr(connectionStr string) (dbName string, dbParams map[string]interface{}, err error) {
+
+	// Derive the database name dbName (sqlite, postgres, mongo etc.)
+	dbNameSep := strings.Index(connectionStr, ":")
+	if dbNameSep == -1 {
+		err = ErrDBConnStrParseNoSep
+		return
+	}
+
+	dbName = strings.TrimSpace(connectionStr[:dbNameSep])
+
+	// Get the parameters to initialize the database
+	dbParams = make(map[string]interface{})
+	for _, param := range strings.Split(connectionStr[dbNameSep+1:], ";") {
+		paramSep := strings.Index(param, "=")
+		if paramSep != -1 {
+			key := strings.TrimSpace(param[:paramSep])
+			if len(key) > 0 {
+				values := strings.Split(param[paramSep+1:], ",")
+				if len(values) > 1 {
+					for i, _ := range values {
+						values[i] = strings.TrimSpace(values[i])
+					}
+					dbParams[key] = values
+				} else {
+					dbParams[key] = strings.TrimSpace(values[0])
+				}
+			}
+		}
+	}
+
+	return
+}
+
+func NewDatabase(connectionStr string) (Database, error) {
+
+	dbName, dbParams, err := parseConnectionStr(connectionStr)
 	if err != nil {
-		return err
+		return nil, err
 	}
-	statement := `
-		CREATE TABLE IF NOT EXISTS comments (
-			url varchar(2083) not null,
-			name varchar(200) not null,
-			comment varchar(3000) not null,
-			depth int not null,
-			time timestamp not null,
-			parent int
-		);
-	`
-	_, err = db.Exec(statement)
-	return err
 
+	switch dbName {
+	case "sqlite":
+		return sqliteInit(dbParams)
+	case "postgres":
+		return nil, ErrDBConnStrParseNotImplemented("postgres")
+	case "mongo":
+		return nil, ErrDBConnStrParseNotImplemented("mongo")
+	default:
+		return nil, ErrDBConnStrParseNotImplemented(dbName)
+	}
 }
 
-func CleanupOldComments() error {
-	statement := `
-		DELETE FROM comments
-		WHERE time < date('now', '-30 minute');
-	`
-	_, err := db.Exec(statement)
+var db Database
+func LoadDatabase(dbFile string) error {
+
+	var err error
+	db, err = NewDatabase("sqlite:file=sqlite3.db")
```

Comment:
Or better yet, let the user give sqlite/postgres/mongo from main through `LoadDatabase`. Basically pass on the variable. It looks more transparent and someone wanting to use postgres can easily figure out how to.